include/experimental/__p1673_bits/blas3_matrix_product.hpp

```diff
@@ -664,56 +1032,116 @@ template<class ElementType_A,
+         class Layout_E,
+         class Accessor_E,
+         class ElementType_C,
+         extents<>::size_type numRows_C,
+         extents<>::size_type numCols_C,
+         class Layout_C,
+         class Accessor_C>
+void symmetric_matrix_left_product(
+  ExecutionPolicy&& /* exec */,
+  std::experimental::mdspan<ElementType_A, std::experimental::extents<numRows_A, numCols_A>, Layout_A, Accessor_A> A,
+  Triangle t,
+  std::experimental::mdspan<ElementType_B, std::experimental::extents<numRows_B, numCols_B>, Layout_B, Accessor_B> B,
+  std::experimental::mdspan<ElementType_E, std::experimental::extents<numRows_E, numCols_E>, Layout_E, Accessor_E> E,
+  std::experimental::mdspan<ElementType_C, std::experimental::extents<numRows_C, numCols_C>, Layout_C, Accessor_C> C)
+{
+  symmetric_matrix_left_product (A, t, B, E, C);
+}
+
+template<class ElementType_A,
+         extents<>::size_type numRows_A,
+         extents<>::size_type numCols_A,
+         class Layout_A,
+         class Accessor_A,
+         class Triangle,
+         class ElementType_B,
+         extents<>::size_type numRows_B,
+         extents<>::size_type numCols_B,
+         class Layout_B,
+         class Accessor_B,
+         class ElementType_E,
+         extents<>::size_type numRows_E,
+         extents<>::size_type numCols_E,
+         class Layout_E,
+         class Accessor_E,
+         class ElementType_C,
+         extents<>::size_type numRows_C,
+         extents<>::size_type numCols_C,
+         class Layout_C,
+         class Accessor_C>
+void symmetric_matrix_right_product(
+  std::experimental::mdspan<ElementType_A, std::experimental::extents<numRows_A, numCols_A>, Layout_A, Accessor_A> A,
+  Triangle /* t */,
+  std::experimental::mdspan<ElementType_B, std::experimental::extents<numRows_B, numCols_B>, Layout_B, Accessor_B> B,
+  std::experimental::mdspan<ElementType_E, std::experimental::extents<numRows_E, numCols_E>, Layout_E, Accessor_E> E,
+  std::experimental::mdspan<ElementType_C, std::experimental::extents<numRows_C, numCols_C>, Layout_C, Accessor_C> C)
+{
+  using size_type = typename extents<>::size_type;
+
+  if constexpr (std::is_same_v<Triangle, lower_triangle_t>) {
+    for (size_type j = 0; j < C.extent(1); ++j) {
+      for (size_type i = j; i < C.extent(0); ++i) {
+        C(i,j) = E(i,j);
+        for (size_type k = 0; k < A.extent(1); ++k) {
+          C(i,j) += B(i,k) * A(k,j);
         }
       }
     }
-    else { // upper_triangle_t
-      for (size_type j = 0; j < C.extent(1); ++j) {
-        for (size_type i = 0; i <= j; ++i) {
-          C(i,j) = E(i,j);
-          for (size_type k = 0; k < A.extent(1); ++k) {
-            C(i,j) += B(i,k) * A(k,j);
-          }
+  }
+  else { // upper_triangle_t
+    for (size_type j = 0; j < C.extent(1); ++j) {
+      for (size_type i = 0; i <= j; ++i) {
+        C(i,j) = E(i,j);
+        for (size_type k = 0; k < A.extent(1); ++k) {
+          C(i,j) += B(i,k) * A(k,j);
```

Comment:
Both the symmetric and Hermitian left and right products should go over the matrix `A` twice, unlike the triangular left and right products.

Reply:
I think we should separate that into its own issue.

Reply:
@amklinv-nnl I might be a bit confused perhaps -- were those functions implemented before?  If not and if this PR provides a partial, known-incorrect implementation for later revision, then would you consider putting `assert(false);` in the functions known to be correct, so that we know to come back to them later?  Thanks! : - )